File: src/main/java/org/cobbzilla/util/io/FilenameSuffixFilter.java
package org.cobbzilla.util.io;

import java.io.File;
import java.io.FilenameFilter;

public class FilenameSuffixFilter implements FilenameFilter {

    private String suffix;

    @java.beans.ConstructorProperties({"suffix"})
    public FilenameSuffixFilter(String suffix) {
        this.suffix = suffix;
    }

    @Override public boolean accept(File dir, String name) {
        return name.endsWith(suffix);
    }

    public String getSuffix() {
        return this.suffix;
    }

    public void setSuffix(String suffix) {
        this.suffix = suffix;
    }
}

File: src/main/java/org/cobbzilla/util/io/FilenameSuffixFilter.kt
package org.cobbzilla.util.io

import java.io.File
import java.io.FilenameFilter

class FilenameSuffixFilter @java.beans.ConstructorProperties("suffix")
constructor(var suffix: String?) : FilenameFilter {

    override fun accept(dir: File, name: String): Boolean {
        return name.endsWith(suffix!!)
    }
}
